sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -285,6 +285,72 @@ case class ArrayTransform(
   override def prettyName: String = "transform"
 }
 
+
+case class ArraySorting(
+    argument: Expression,
+    function: Expression)
+  extends ArrayBasedSimpleHigherOrderFunction with CodegenFallback {
+
+
+  @transient lazy val elementType: DataType =
+    argument.dataType.asInstanceOf[ArrayType].elementType
+  override def dataType: ArrayType = ArrayType(elementType, function.nullable)
+
+  override def bind(f: (Expression, Seq[(DataType, Boolean)]) => LambdaFunction): ArraySorting = {
+    val ArrayType(elementType, containsNull) = argument.dataType
+    function match {
+      case LambdaFunction(_, arguments, _) if arguments.size == 2 =>
+        copy(function =
+          f(function, (elementType, containsNull) :: (elementType, containsNull) :: Nil))
+      case _ =>
+        copy(function = f(function, (elementType, containsNull) :: Nil))
+    }
+  }
+
+  @transient lazy val (firstParam, secondParam) = {
+    val LambdaFunction(_, (firstParam: NamedLambdaVariable) +: tail, _) = function
+    val secondParam = tail.head.asInstanceOf[NamedLambdaVariable]
+    (firstParam, secondParam)
+  }
+
+
+  override def nullSafeEval(inputRow: InternalRow, argumentValue: Any): Any = {
+    val arr = argumentValue.asInstanceOf[ArrayData]
+    val f = functionForEval
+
+    @transient lazy val customComparator: Comparator[Any] = {
+      (o1: Any, o2: Any) => {
+        if (o1 == null && o2 == null) {
+          0
+        } else if (o1 == null) {
+          1
+        } else if (o2 == null) {
+          -1
+        } else {
+          firstParam.value.set(o2)
+          secondParam.value.set(o1)
+          f.eval(inputRow).asInstanceOf[Int]
+        }
+      }
+    }
+
+   sortEval(arr, customComparator)
+  }
+
+  def sortEval(array: Any, comparator: Comparator[Any]): Any = {
+    val data = array.asInstanceOf[ArrayData].toArray[AnyRef](elementType)
+    if (elementType!= NullType) {
+      java.util.Arrays.sort(data, comparator)
+    }
+    new GenericArrayData(data.asInstanceOf[Array[Any]])
+  }
+
+  override def prettyName: String = "array_new_sort"
```

Comment:
Why is it a new name instead of just `arrow_sort`?

Reply:
True, it was for testing purposes. Done